awx/settings/development.py

```diff
@@ -105,8 +92,22 @@
 AWX_DISABLE_TASK_MANAGERS = False
 # ======================!!!!!!! FOR DEVELOPMENT ONLY !!!!!!!=================================
 
-if 'sqlite3' not in DATABASES['default']['ENGINE']:  # noqa
-    DATABASES['default'].setdefault('OPTIONS', dict()).setdefault('application_name', f'{CLUSTER_HOST_ID}-{os.getpid()}-{" ".join(sys.argv)}'[:63])  # noqa
+# Store a snapshot of default settings at this point before loading any
+# customizable config files.
+this_module = sys.modules[__name__]
+local_vars = dir(this_module)
+DEFAULTS_SNAPSHOT = {}
+for setting in local_vars:
+    if setting == setting.upper():
+        DEFAULTS_SNAPSHOT[setting] = copy.deepcopy(getattr(this_module, setting))
+
+del local_vars
+del this_module
```

Comment:
I wanted to `del` these because we do use `dir` on the settings module in our settings wrapper:

https://github.com/ansible/awx/blob/devel/awx/conf/settings.py#L474

and I just don't want to pollute that namespace with more variables that should be internal to logic inside of settings. It would be impractical to add an `__ALL__` to this module. Also, the original recursive copy here was because of lack of awareness of what things hanging out in locals(). We avoid copying lowercase vars in the loop, but I prefer to explicitly exclude them from the namespace, as they should not be public.